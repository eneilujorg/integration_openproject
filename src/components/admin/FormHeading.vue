<template>
	<div class="form-heading"
		:class="{'disabled': isDisabled}">
		<div v-if="isComplete" class="complete">
			<div class="icon icon-checkmark" />
		</div>
		<div v-else class="index">
			{{ translate(index) }}
		</div>
		<div class="title"
			:class="{'green-text': isComplete}">
			{{ translate(title) }}
		</div>
	</div>
</template>
<script>
import { translate as t } from '@nextcloud/l10n'

export default {
	name: 'FormHeading',
	props: {
		index: {
			type: String,
			required: true,
		},
		title: {
			type: String,
			required: true,
		},
		isDisabled: {
			type: Boolean,
			default: false,
		},
		isComplete: {
			type: Boolean,
			default: false,
		},
	},
	methods: {
		translate(text) {
			return t('integration_openproject', text)
		},
	},
}
</script>
<style lang="scss" scoped>
.form-heading {
	display: flex;
	justify-content: start;
	align-items: center;
	padding: 20px 0;

	.green-text {
		color: #308720;
	}

	.complete {
		height: 16px;
		width: 16px;
		border-radius: 50%;
		background: #308720;
		.icon {
			filter: invert(1);
			background-size: 12px;
		}
	}

	.index {
		height: 16px;
		width: 16px;
		font-size: 14px;
		font-weight: 600;
		line-height: 16px;
		text-align: center;
		border-radius: 50%;
		background: #555555;
		color: white;
	}
	.title {
		font-weight: 700;
		font-size: 14px;
		line-height: 20px;
		padding-left: 6px;
	}
}

.form-heading.disabled {
	.index {
		background: #CCCCCC !important;
	}
	.title {
		color: #CCCCCC !important;
	}
}
</style>
