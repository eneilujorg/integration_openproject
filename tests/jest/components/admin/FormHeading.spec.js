/**
 * SPDX-FileCopyrightText: 2022-2023 Nextcloud GmbH and Nextcloud contributors
 * SPDX-FileCopyrightText: 2022 Jankari Tech Pvt. Ltd.
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import { createLocalVue, mount } from '@vue/test-utils'
import FormHeading from '../../../../src/components/admin/FormHeading.vue'

const localVue = createLocalVue()

global.t = (app, text) => text

describe('FormHeading.vue', () => {
	describe('is complete prop', () => {
		it('should show checkmark icon, add green title and hide the index if complete', () => {
			const wrapper = getWrapper({
				isComplete: true,
			})
			expect(wrapper).toMatchSnapshot()
		})
		it('should hide the checkmark icon and show the index if not complete', () => {
			const wrapper = getWrapper({
				isComplete: false,
			})
			expect(wrapper).toMatchSnapshot()
		})
	})
	describe('is disabled prop', () => {
		it('should add disabled class to the form heading', () => {
			const wrapper = getWrapper({
				isDisabled: true,
			})
			expect(wrapper).toMatchSnapshot()
		})
	})
})

function getWrapper(props = {}) {
	return mount(FormHeading, {
		localVue,
		propsData: {
			title: 'Some Field Title',
			index: '1',
			...props,
		},
		mocks: {
			t: (app, msg) => msg,
		},
	})
}
