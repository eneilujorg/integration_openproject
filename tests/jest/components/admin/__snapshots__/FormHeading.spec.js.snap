// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHeading is complete prop should hide the checkmark icon and show the index if not complete 1`] = `
<div class="form-heading">
  <div class="index">
    1
  </div>
  <div class="title">
    Some Field Title
  </div>
</div>
`;

exports[`FormHeading is complete prop should show checkmark icon, add green title and hide the index if complete 1`] = `
<div class="form-heading">
  <div class="complete"><span aria-hidden="true" role="img" class="material-design-icon check-bold-icon"><svg fill="#FFFFFF" width="12" height="12" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"><!----></path></svg></span></div>
  <div class="title green-text">
    Some Field Title
  </div>
</div>
`;

exports[`FormHeading is disabled prop should add disabled class to the form heading 1`] = `
<div class="form-heading disabled">
  <div class="index">
    1
  </div>
  <div class="title">
    Some Field Title
  </div>
</div>
`;
